src/java.desktop/share/classes/javax/swing/ImageIcon.java

```diff
@@ -105,39 +105,37 @@
      * It is left for backward compatibility only.
      * @deprecated since 1.8
      */
+    @SuppressWarnings("removal")
     @Deprecated
-    protected static final Component component;
+    protected static final Component component
+            = AccessController.doPrivileged(new PrivilegedAction<Component>() {
+        public Component run() {
+            try {
+                final Component component = createNoPermsComponent();
+
+                // 6482575 - clear the appContext field so as not to leak it
+                AWTAccessor.getComponentAccessor().
+                        setAppContext(component, null);
+
+                return component;
+            } catch (Throwable e) {
+                // We don't care about component.
+                // So don't prevent class initialisation.
+                e.printStackTrace();
```

Comment:
I am not the reviewer, but I am curious. If we do not care about the exception, is it a right way to just print it to the console as it is? If people use loggers, they won’t be able to capture this stack trace.
thank you for your answer in advance.

Reply:
No. A refactoring change like this is the wrong time to make any unrelated changes. If such changes are a good idea, they should be done in a different bug with a separate pull request.